core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -573,9 +574,9 @@ class GroupMetadataManager(brokerId: Int,
                     } else {
                       val offsetAndMetadata = GroupMetadataManager.readOffsetMessageValue(record.value)
                       if (isTxnOffsetCommit)
-                        pendingOffsets(batch.producerId).put(groupTopicPartition, CommitRecordMetadataAndOffset(Some(batch.baseOffset), offsetAndMetadata))
+                        pendingOffsets(batch.producerId).put(groupTopicPartition, CommitRecordMetadataAndOffset(Some(batchBaseOffset), offsetAndMetadata))
                       else
-                        loadedOffsets.put(groupTopicPartition, CommitRecordMetadataAndOffset(Some(batch.baseOffset), offsetAndMetadata))
+                        loadedOffsets.put(groupTopicPartition, CommitRecordMetadataAndOffset(Some(batchBaseOffset), offsetAndMetadata))
```

Comment:
So I think the reason we use the base offset is for consistency with the append path, but I don't think it's strictly necessary. Especially considering __consumer_offsets is compacted, the base offset is changing all the time anyway (for older message format versions). The only guarantee we really need is that we can compare the order of two offset commits using the offset of the corresponding messages. 

That said, it's fine with me to leave this for a future patch to keep the risk here low. To avoid the second decompress, grabbing the base offset during the iteration as suggested sounds good to me.

Reply:
`val batchBaseOffset = Some(batch.baseOffset)` ?